### DISCUSSION TITLE
How can I replace dashes/hyphens in flag names with underscores?

### DISCUSSION THREAD

**Discussion by sagikazarmark** [spf13/viper#1054] [Q&A] [answered]
Flags are usually in the form of `this-is-a-flag` whereas Viper keys are usually dot notated.

**Comment by sagikazarmark** [comment, marked as answer]
Answered by @megapctr in https://github.com/spf13/viper/issues/1044#issue-759509428:

> I found a solution, this issue is just for folks who may also be struggling with this. Feel free to close it, unless you think it's worth mentioning in the readme.
> 
> I wanted to keep my CLI flags in `kebab-case` and my config keys in `snake_case`. I've been searching for something like `EnvKeyReplacer` but for flags. Turns out `pflag.FlagSet` has a function for normalizing flag names.
> 
> This solved my issue:
> 
> ```go
> // bind cmd flags to viper, replacing dashes with underscores
> f := cmd.Flags()
> normalizeFunc := f.GetNormalizeFunc()
> f.SetNormalizeFunc(func(fs *pflag.FlagSet, name string) pflag.NormalizedName {
>     result := normalizeFunc(fs, name)
>     name = strings.ReplaceAll(string(result), "-", "_")
>     return pflag.NormalizedName(name)
>  })
>  viper.BindPFlags(f)
> ```



> I found a solution, this issue is just for folks who may also be struggling with this. Feel free to close it, unless you think it's worth mentioning in the readme.
> 
> I wanted to keep my CLI flags in `kebab-case` and my config keys in `snake_case`. I've been searching for something like `EnvKeyReplacer` but for flags. Turns out `pflag.FlagSet` has a function for normalizing flag names.
> 
> This solved my issue:
> 
> ```go
> // bind cmd flags to viper, replacing dashes with underscores
> f := cmd.Flags()
> normalizeFunc := f.GetNormalizeFunc()
> f.SetNormalizeFunc(func(fs *pflag.FlagSet, name string) pflag.NormalizedName {
>     result := normalizeFunc(fs, name)
>     name = strings.ReplaceAll(string(result), "-", "_")
>     return pflag.NormalizedName(name)
>  })
>  viper.BindPFlags(f)
> ```

**Comment by ffalor** [comment]
~~When I do this for persistent flags the flag names are changed breaking subcommands. Does this only work for regular flags or am I doing something wrong.~~

Nvm.. just not going to use cobra lol.

Thanks!
